### DISCUSSION TITLE
Toward a better localizing experience

### DISCUSSION THREAD

**Discussion by Glandos** [element-hq/element-meta#2134] [Product Feedback]
Hi!

First of all, small background: I'm primarily a developer, and sometimes a translator, as I am for Element Web.

I was quite happy with the experience on https://translate.element.io/ The UI is less modern than some other translation platforms, but efficient, and that seems the main goal of a translation tool for me. Get the job done in an efficient way.

I was notified of the move to Localazy for Element Web (tracked in https://github.com/vector-im/element-web/issues/25608), due to technical requirements from the Element X team. I am not really a fan of proprietary platform for translations of open source projects, but the UX seems to be smoother than the Weblate-based one, so I was happy to give it a try.

However, here comes the paint points. It's basically the same content I posted in https://github.com/vector-im/element-web/issues/25608#issuecomment-1754725395:

- Onboarding is difficult: create a localazy account, ask access to language in [matrix.to/#/#element-translations:matrix.org](https://matrix.to/#/#element-translations:matrix.org) then accept invitation, then re-notify room to get access.
- Access to other languages is forbidden: it seems that localazy has no "read-only" access. This really helpful to see the current state of translations, what other translators did to solve some issue
- Keyboard shortcut are far from complete: to fill a suggestion, it is required to blur translation input with ESC, then insert with SHIFT+1 (or 2, 3, etc.). But refocusing input can only be done either with tabbing like a monkey, since / triggers my browser find feature instead of focusing input.
- Keyboard shortcut, reloaded: placeholders, such as %(value) are hard to type. In Weblate, shortcuts CTRL+1 (2, 3 for more) insert placeholder at cursor location. It seems only possible to click on the placeholder in Localazy, this is a hard TMS trigger.

As a developer, the most striking fact is the nonexistence of a read-only access. As a translator, the most painful fact is the lack of shortcut for placeholders, since they are usually hard to type (on purpose).

I don't ask for an explanation on why this platform was used, since it fits a developer requirement, to avoid duplication, to bring a better coherence to Element product family.
I open this discussion to any ideas that can improve or remove those pain points. If you have answers, workarounds, or even proper fixes for them, I'll be delighted to hear (or *read*) them.

EDIT: Now, Localazy has enshittified to the point of doing illegal tracking according to GDPR: https://github.com/element-hq/element-meta/discussions/2134#discussioncomment-14814961 it seems that Element should move away from this platform. It's not even debatable from my point of view.

**Comment by meskobalazs** [comment]
I would also add one pain point. Localazy does not have offline translation capabilities, which is a massive hit in productivity. The support of PO file generation, and updating from them is one of the best features of Weblate.

**Reply by Glandos** [reply]
This is not a real answer, but I've successfully been able to use [Poedit](https://github.com/vslavik/poedit) from it's latest release (3.4). It's not offline, it's not PO :)

**Reply by meskobalazs** [reply]
Chiming in more than one year later. This worked quite well for ElementX, where I am a _Trusted Translator_, but it's basically unusable in Element Web, where I can only suggest new translations.

**Comment by Glandos** [comment]
https://github.com/vector-im/element-meta/assets/297578/7ad1fa50-6139-4dc0-a569-8bc20ad73d0c

This is another issue. It seems that project authorization are not really consistent.

**Comment by Glandos** [comment]
When a change occurred in source language, Weblate used to show this change as a diff. In Localazy, when only have a notification in the translation input, without knowing what to change:
![image](https://github.com/vector-im/element-meta/assets/297578/7eab40a2-cb69-4e2a-8ecc-20202483ebdd)

**Comment by Glandos** [comment]
I don't know how to get more visibility from ElementX team, but this discussion is currently in [the top 4](https://github.com/vector-im/element-meta/discussions?discussions_q=is%3Aopen+sort%3Atop+created%3A%3E%3D2022-10-26).
So I'm pinging @bmarty @stefanceriu @pixlwave as gently as possible (Top contributors on [ElementX Android](https://github.com/vector-im/element-x-android) and [ElementX iOS](https://github.com/vector-im/element-x-ios) and [owners in Localazy](https://localazy.com/p/element-web/contributors?sort=-1))

**Reply by pixlwave** [reply]
Hey, thanks for opening this discussion. For the first 2 points, we have changed the access permissions on Element Web (and will do similar on Element X) to make the project more open.

For the second 2 points on keyboard shortcuts, those sound like valid feedback to me, but is very much feedback for Localazy rather than ourselves. I would suggest you contact them with suggestions on what would make translation easier for you. Their email address is here: https://localazy.com/contact

Similar thoughts for the diff on changes to the source string.

**Reply by Glandos** [reply]
Thanks for taking care of that! The openness of this project is rather important for me :+1: 

Speaking of that, I won't take the time to contact Localazy. I don't have any interest in improving their product. I deeply think that FLOSS is the way to go for tooling.

**Reply by Glandos** [reply]
Plus, I have searched in their forum for keyboard shortcuts to insert placeholders: https://discuss.localazy.com/t/shortcut-for-inserting-placeholders-and-markup/274

Idea posted and validated in 2021. Nothing since then. It seems like a waste of my resources to contact them.

By the way, I stopped translating Element in French, this is too difficult from the user experience right now.

**Comment by Glandos** [comment]
It's not really obvious on this capture, but warnings from rules display a dialog on validation. This is OK, but if I press `Escape`, the focus is not in the translation input anymore.


https://github.com/vector-im/element-meta/assets/297578/7346902a-c287-4aa8-bfd9-34195d5dfe95

This was also an issue in the Weblate used by Element, but it was fixed in recent version of Weblate.

**Reply by pixlwave** [reply]
Similarly to https://github.com/vector-im/element-meta/discussions/2134#discussioncomment-7467264, this is an issue to take up with Localazy 🙂

**Comment by Glandos** [comment]
I'm unable to search for keys or string in source language. In my current use case, I wanted to know where "knock" were used and translated to have a consistent translation.
Is there a way to have such an overview?

**Reply by t3chguy** [reply]
@Glandos is your complaint specific to https://localazy.com/p/element - does it apply to https://localazy.com/p/element-web?

**Reply by Glandos** [reply]
It was about https://localazy.com/p/element-web
As far as I understood, I don't have access to https://localazy.com/p/element (Element X), since the team would like to hire professional translators for (at least) French.

**Reply by t3chguy** [reply]
Hmm yes, interesting that they lock down the source language like that.

**Comment by erkinalp** [comment]
I would prefer a special-purpose Matrix room version, where every base string is a state event (`m.translation_key`), and translated strings are state events with two-dimensional state keys (`m.translation_key`+`lang`).

**Reply by t3chguy** [reply]
How would review & permissions work for such a system? Can't just give everyone write access to all languages.

**Reply by erkinalp** [reply]
On a RBAC basis per-language **and** per-translation-key, with defaulting to read only on both axises.

**Reply by t3chguy** [reply]
That isn't something Matrix state supports, so you'd need a room per translation key, which seems subpar.

**Reply by erkinalp** [reply]
Currently existing room versions do not support two-dimensional state keys, but there is nothing intrinsic to how Matrix works that would prevent such a room version from being authored.

**Comment by Glandos** [comment]
EDIT: It's now possible to reject all in one click, see below.

ORIGINAL TEXT:
This has become unacceptable:
<img width="896" height="390" alt="image" src="https://github.com/user-attachments/assets/6c43deb9-f63d-48fd-8769-0e0ca628a27d" />
And clicking on `EDIT PREFERENCES`:
<img width="896" height="944" alt="image" src="https://github.com/user-attachments/assets/490461c0-a81a-4b77-b1b1-8e8dfb5f7a82" />

This clearly violates GDPR, since it's opt out, and I'm using a EU IP address.

I kindly ask once again to use another translation platform. This one is following the enshittification pattern too quickly.

**Reply by Glandos** [reply]
For reference: https://gdpr.eu/cookies/#:~:text=users%20more%20broadly.-,cookie%20compliance,-To%20comply%20with
> To comply with the regulations governing cookies under the GDPR and the ePrivacy Directive you must:
    - Receive users’ consent before you use any cookies except strictly necessary cookies.
    - Provide accurate and specific information about the data each cookie tracks and its purpose in plain language before consent is received.
    - Document and store consent received from users.
    - Allow users to access your service even if they refuse to allow the use of certain cookies
    - Make it as easy for users to withdraw their consent as it was for them to give their consent in the first place.

**Reply by Glandos** [reply]
Other references
https://www.edpb.europa.eu/system/files/2023-01/edpb_20230118_report_cookie_banner_taskforce_en.pdf

> When authorities were asked whether they would consider that a banner which does not provide for
accept and refuse/reject/not consent options on any layer with a consent button is an infringement of
the ePrivacy Directive, a vast majority of authorities considered that the absence of refuse/reject/not
consent options on any layer with a consent button of the cookie consent banner is not in line with the
requirements for a valid consent and thus constitutes an infringement. Few authorities considered that
they cannot retain an infringement in this case as article 5(3) of the ePrivacy Directive does not
explicitly mentioned a “reject option” to the deposit of cookies

> The taskforce members confirmed that pre-ticked boxes to opt-in do not lead to valid consent as
referred to either in the GDPR (see in particular recital 32 “Silence, pre-ticked boxes or inactivity should
not therefore constitute consent.”) or in Article 5(3) of the ePrivacy Directive.

**Reply by erkinalp** [reply]
hence my suggestion to have a custom matrix room version

**Reply by Glandos** [reply]
For the record, I asked about it using the chatbot of Localazy. Here is the complete transcript:
```
Conversation with Localazy
Started on October 29, 2025 at 01:59 PM Europe/Prague time CET (GMT+0100)

---

01:59 PM | Glandos: Hi,
I'm a user of your platform, since Element Web & X decided to use it.
I discovered today the cookie banner, and I have to say that it's non compliant with GDPR , affecting EU residents.
Basically, tracking is enabled by default, and there is no "REJECT ALL" button on the same layer as "ACCEPT ALL". 
 Do you have plan to make it compliant?

02:02 PM | Operator: You’ll get replies here and in your email:
 Our usual reply time
🕒 under 30 minutes

02:02 PM | Operator: Are you a Localazy customer?

02:05 PM | Glandos: No, I'm not a customer yet

02:05 PM | Operator: Welcome! You'll get a reply from the team.

02:05 PM | Operator: What you're looking for?

02:06 PM | Glandos: Just a question

02:06 PM | Operator: A few more details will help get you to the right person:

02:06 PM | Operator: Thanks! The team will reply as soon as they can.

03:06 PM | Petr from Localazy: Hi, thank you for the report. I have tasked our marketing team to update the cookie consent.

---
Exported from Localazy on October 29, 2025 at 03:14 PM Europe/Prague time CET (GMT+0100)
```

**Reply by Glandos** [reply]
They've added a `REJECT ALL` button, which is nice.
<img width="896" height="366" alt="Screenshot 2025-11-20 at 15-13-21 Translate – Element Web   Desktop" src="https://github.com/user-attachments/assets/50e56234-4972-4763-bf47-c936c53abd62" />

However, when going to `EDIT PREFERENCES`, optional trackers are enabled by default. But this is still a good point.

**Comment by pixlwave** [comment]
To provide more context to https://github.com/element-hq/element-x-ios/issues/4673:

We decided to move away from Weblate for Element X on both platforms collectively as we had the following requirements which Weblate couldn't meet:
- Use a single project for both iOS (.strings & .stringsdict) and Android (.xml) strings so that we don't duplicate the localisation work and ensure that the same string values are used on both platforms.
- Good support for plurals on iOS (weblate required 1 project for .strings and another for .stringsdict which is subpar).
- English strings need to be added within the web UI and easy to edit so that Product/Design can update them without requiring engineering time.
- To have the platform work in a download-only way where it is the source of truth and any conflicts can be fixed by downloading the latest strings.
    - Both EI and EA have specific rules for how to sync the project with weblate otherwise conflicts occur which can be hard to resolve and in have resulted in the loss of translators work when a mistake was made.

Whilst these were the base requirements at the time, Localazy has also given us
- The ability to review changes before accepting them (and to trust this to community members when we see strong contributions from specific people)
    - The backwards machine translation part of this is also fantastic and avoids the need to round-trip to some other translation service when checking strings.
- Support for duplicity management both within the project and across multiple projects (although we're yet to adopt the latter).
- A much easier to use UI.
- A built in comments system allowing translators to ask for clarifications directly on specific strings.

Whilst Localazy does have its downsides (for us as well – I'd love the ability to e.g. easily see which languages have active reviewers), the upsides are far greater right now. I have suggested in the [past](https://github.com/element-hq/element-meta/discussions/2134#discussioncomment-7467264) to raise your concerns directly with Localazy and can only suggest the same thing again.

**Reply by erkinalp** [reply]
why not a purpose-built room version with multi-dimensional state keys?

**Reply by t3chguy** [reply]
Contributions welcome, if you get all the necessary MSCs landed along with a high quality translation contribution & review tool with a lot of similar features from Localazy then it'd be hard to resist surely. Spending weeks/months developing our own tool and then maintaining it into the permanent future is not something we have the personnel to staff currently, without needing to neglect other projects.

**Reply by Glandos** [reply]
As a developer myself, I totally understand the importance of a choosing a tool that is built for the use case.
In this case, Localazy seems to be built for product managers and developers.

On the other hand, when opt-out advertisement in there, the product is not the tool anymore, it's user data. Having this kind of business model should trigger a warning on whether or not the team want to be locked-in by a company that starts to shift its revenue from SaaS to advertisement.

I thought that the libre/open source spirit was important for Element teams, because of the mission that is part of the underlying protocol.

Finally: no, I won't try to contact Localazy for feature improvement. I would love to do it for Weblate. I've contacted Localazy support about the cookie consent banner, because it seems illegal.
